Ooui/Node.cs

```diff
@@ -117,10 +117,77 @@ protected void ReplaceAll (Node newNode)
             foreach (var child in toRemove) {
                 child.MessageSent -= HandleChildMessageSent;
                 Call ("removeChild", child);
+                // TODO: Should call OnChildRemoved?
             }
             InsertBefore (newNode, null);
         }
 
+        public void ReplaceChildrenWith (params Node[] nodes)
+        {
+            ReplaceChildren(nodes);
+        }
+
+        public void ReplaceChildren (IEnumerable<Node> replaceWith)
+        {
+            var wanted = new List<Node> (16);
```

Comment:
A validation that lacks here (and in other places AFAICT) is if the node already has a parent. I am not sure what the wanted behavior in this case is so I didn't put in code for it.